ambry-router/src/test/java/com.github.ambry.router/DeleteManagerTest.java

```diff
@@ -353,21 +328,19 @@ public void testVariousServerErrorCodesForThreeParallelism()
   public void testResponseTimeout()
       throws Exception {
     setServerResponse(false);
-    ServerErrorCode[] serverErrorCodes = new ServerErrorCode[9];
-    Arrays.fill(serverErrorCodes, ServerErrorCode.No_Error);
-    presetServerErrorCode(serverErrorCodes);
-    future = router.deleteBlob(blobIdString, new ClientCallback());
-    do {
-      // increment mock time
-      mockTime.sleep(1000);
-    } while (!operationCompleteLatch.await(10, TimeUnit.MILLISECONDS));
-    try {
-      future.get();
-      fail("Deletion should be unsuccessful. Exception is expected.");
-    } catch (Exception e) {
-      assertEquals("RouterErrorCode should be OperationTimedOut", RouterErrorCode.OperationTimedOut,
-          ((RouterException) e.getCause()).getErrorCode());
-    }
+    testWithErrorCodes(Collections.singletonMap(ServerErrorCode.No_Error, 9), serverLayout,
+        RouterErrorCode.OperationTimedOut, new ErrorCodeChecker() {
+          @Override
+          public void check(RouterErrorCode expectedError)
+              throws Exception {
+            future = router.deleteBlob(blobIdString, new ClientCallback());
+            do {
+              // increment mock time
+              mockTime.sleep(1000);
+            } while (!operationCompleteLatch.await(10, TimeUnit.MILLISECONDS));
```

Comment:
So it seems like this gets called multiple (6) times even though, the timeout is set to 1 second. @mxia, do you know why this is the case?